File: social-map-fe/src/services/searchHistoryService.js
import { api } from './apiClient';

export const searchHistoryService = {
    /**
     * Lưu lịch sử tìm kiếm
     * POST /api/search-history
     */
    saveSearchHistory: async (query, type, data = null) => {
        return await api.post('/search-history', { query, type, data });
    },

    /**
     * Lấy lịch sử tìm kiếm của người dùng (dựa vào JWT)
     * GET /api/search-history
     */
    getSearchHistory: async () => {
        return await api.get('/search-history');
    },

    /**
     * Xóa toàn bộ lịch sử tìm kiếm của người dùng (dựa vào JWT)
     * DELETE /api/search-history
     */
    deleteSearchHistory: async () => {
        return await api.delete('/search-history');
    },

    /**
     * Xóa một lịch sử tìm kiếm cụ thể
     * DELETE /api/search-history/{id}
     */
    deleteSearchHistoryItem: async (id) => {
        return await api.delete(`/search-history/${id}`);
    },
};

File: social-map/src/main/java/com/mapsocial/service/SearchHistoryService.java
package com.mapsocial.service;

import com.mapsocial.entity.SearchHistory;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface SearchHistoryService {

    void saveSearchHistory(UUID userId, String query, String type, String data);

    List<SearchHistory> getSearchHistory(UUID userId);

    List<SearchHistory> getTop5SearchHistory(UUID userId);

    void deleteSearchHistory(UUID userId);

    Optional<SearchHistory> findById(Long id);

    void delete(SearchHistory history);
}
